src/background/adblocker.js

```diff
@@ -196,6 +241,21 @@ async function injectScriptlets(scripts, tabId, frameId) {
     script.remove();
   }
 
+  if (__PLATFORM__ === 'firefox') {
+    if (scripts.length === 0) {
+      contentScripts.unregister(hostname);
```

Comment:
I'm wondering if this part of the code is necessary or how it should be best handled. 
When the engine updates, it should automatically clear the content scripts (contentScripts.unregisterAll). There might be potential for race conditions, but in principle, it should eventually take care of unregistering.

Is the code here intended especially for pause (which won't trigger an engine reload)? If so, would be good to leave a comment explaining this.